src/backend/joanie/payment/api.py

```diff
@@ -62,5 +60,9 @@ class CreditCardViewSet(
 
     def get_queryset(self):
         """Custom queryset to get user's credit cards"""
-        user = User.update_or_create_from_request_user(request_user=self.request.user)
-        return user.credit_cards.all()
+        username = (
+            self.request.auth["username"]
+            if self.request.auth
+            else self.request.user.username
```

Comment:
In some case you are checking if `self.request.user.is_authenticated`, why not doing this in viewsets? Or why not remove those checks elsewhere?

Reply:
I didn't check here because the permission is `IsAuthenticated`
The tests will fail if someone modifies the permission and allows anonymous users to arrive here :man_shrugging:

Reply:
Yup right!